app/views/shared/_error_summary.html.erb

```diff
@@ -0,0 +1,28 @@
+<% class_name = class_name.presence || object.class.name.demodulize.underscore.humanize.downcase %>
+
+<% if object.errors.present? %>
+  <div class="gem-c-error-summary govuk-error-summary" data-module="govuk-error-summary" aria-labelledby="error-summary-title" role="alert" id="error-summary" data-govuk-error-summary-module-started="true">
```

Comment:
It's a shame that you've had to switch out the partial for a manual HTML implementation of it. We'll fall out of sync if any changes are made in the components gem.

I'm tempted to suggest that we keep using the component from the gem, but additionally loop through the `object.errors` ourselves for the sake of GA tracking. But this alternative approach doesn't come without its own drawbacks.

For example, we'd need to create a bunch of hidden/empty HTML elements – one for each error – so that we can add:
```
data: track_analytics_data("form-error", analytics_action, error_message)
```

for the [AutoTrackEvent](https://github.com/alphagov/govuk_publishing_components/blob/main/app/assets/javascripts/govuk_publishing_components/analytics/auto-track-event.js) JS module still kicks in and tracks the events for us.

Or alternatively we might end up needing to implement some custom JS which would track them for us.

e.g. an inline script which includes every error

```html
<script>
window.GOVUK.analytics.trackEvent(category, action, options)
</script>
```

### TL;DR

It doesn't feel great that we aren't using the error summary partial provided by the gem. But the other alternatives also feel a bit fiddly. 🤷🏻

Reply:
Yea the other option is me manually add it to each input for each view, but not all components in the publishing gems have the ability to it seems better to catch it here.

If the component did anything complex i'd agree, but it's something that i've tweaked myself pretty significantly to link errors correctly etc.

It really doesn't do a tonne tbh. 

The main thing is it fall out of sync with styling though. I'm not overly worried though tbh. Wdyt?

Reply:
We've talked about this and agreed to go ahead with the approach as-is.